packages/nx/src/plugins/js/utils/typescript.spec.ts

```diff
@@ -0,0 +1,23 @@
+import { readTsConfigOptions } from './typescript';
+import { mkdtemp, writeFile } from 'node:fs/promises';
+import { tmpdir } from 'os';
+import { join } from 'path';
+
+describe.only('readTsConfigOptions', () => {
+  it('should handle extends', async () => {
+    const dir = await mkdtemp(join(tmpdir(), 'typescript.spec'));
+    await writeFile(
+      join(dir, 'a.json'),
+      JSON.stringify({ extends: './b.json' })
+    );
+    await writeFile(
+      join(dir, 'b.json'),
+      JSON.stringify({ compilerOptions: { strict: true } })
+    );
```

Comment:
Let's use the `TempFs` class exported by `packages/nx/src/internal-testing-utils.ts`. You can see an example of how to use it at `packages/nx/src/config/to-project-name.spec.ts`.

Reply:
Done